src/intellisense/notebookPythonPathService.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { extensions, Uri } from 'vscode';
+import { IInteractiveWindowProvider } from '../interactive-window/types';
+import { findAssociatedNotebookDocument } from '../notebooks/helpers';
+import { INotebookControllerManager } from '../notebooks/types';
+import { IExtensionSingleActivationService } from '../platform/activation/types';
+import { IPythonApiProvider } from '../platform/api/types';
+import { IVSCodeNotebook } from '../platform/common/application/types';
+import { PylanceExtension, PythonExtension } from '../platform/common/constants';
+import { getFilePath } from '../platform/common/platform/fs-paths';
+import { IConfigurationService } from '../platform/common/types';
+import { IInterpreterService } from '../platform/interpreter/contracts';
+import * as semver from 'semver';
+import { traceInfo, traceVerbose } from '../platform/logging';
+
+/**
+* Manages use of the Python extension's registerJupyterPythonPathFunction API which
+* enables us to provide the python.exe path for a notebook as required for Pylance's
+* LSP-based notebooks support.
+*/
+@injectable()
+export class NotebookPythonPathService implements IExtensionSingleActivationService {
+    private _isEnabled: boolean | undefined;
+
+    constructor(
+        @inject(IPythonApiProvider) private readonly apiProvider: IPythonApiProvider,
+        @inject(IVSCodeNotebook) private readonly notebooks: IVSCodeNotebook,
+        @inject(IInteractiveWindowProvider) private readonly interactiveWindowProvider: IInteractiveWindowProvider,
+        @inject(INotebookControllerManager) private readonly notebookControllerManager: INotebookControllerManager,
+        @inject(IInterpreterService) private readonly interpreterService: IInterpreterService,
+        @inject(IConfigurationService) private readonly configService: IConfigurationService
+    ) {}
+
+    public async activate() {
+        if (!this.isPylanceUsingLspNotebooks()) {
```

Comment:
I think this needs to register the path function regardless of whether or not the experiment is active. The changes you made in the intellisenseProvider.ts handle the case where the experiment setting is updated by the user, but if this class doesn't always register, a change made by the user won't be picked up.

Reply:
I talked with Luciana and she felt that listening for settings changes was not necessary. Anyone that manually changes the setting can deal with restarting VS Code on their own. So I removed the settings watching code in `intellisenseProvider.ts`.

Also, I found if I remove the `isPylanceUsingLspNotebooks` check in `activate`, if Python is not installed, the `PythonApiProvider.getApi()` call causes Jupyter to display a prompt saying: "The Python extension is required to perform that task. Click Yes to open Python extension installation page." So I left that check in.